### PR TITLE
Telephony state support

### PR DESCRIPTION
**Pull request by skynetua**
Fixes: https://github.com/home-assistant/android/issues/441

This PR adds support of telephony states requested at #441 

There is a Phone State sensor added. It may have three states:
idle - no active calls at that moment
ringing - for incoming call
offhook - basically for outgoing call

image

I've made it optional through settings. I think a lot of people don't like idea of "something" tracking their calls.

There is a few things I want to add and need recommendation how to do it better.
1. Register broadcast receiver in application class after initial installation when permissions hasn't yet granted.
2. Register\unregister receiver after corresponding setting change

### CODE REVIEW

**Comment by homeassistant** [comment]
Hi @skynetua, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;android#731).

Once you do that we will be able to review and accept this pull request.

Thanks!

File: app/src/main/res/layout/fragment_mobile_app_integration.xml [hunk 1]
```diff
@@ -140,6 +140,35 @@
...
+            <androidx.appcompat.widget.AppCompatTextView
+                style="@style/TextAppearance.HomeAssistant.Title"
+                android:layout_width="match_parent"
+                android:layout_height="wrap_content"
+                android:layout_marginTop="@dimen/activity_margin"
+                android:text="@string/permission_explanation_calls" />
+
+            <androidx.appcompat.widget.AppCompatButton
+                android:id="@+id/phone_state_perms"
+                style="@style/Widget.HomeAssistant.Button.Colored"
```
Comment (dshokouhi) [inline app/src/main/res/layout/fragment_mobile_app_integration.xml:152]: We may or may not need to remove this, can you test locally with both dark mode and light mode enabled in the system UI? If the button is missing in light mode just remove this line and it should correct it.

**Comment by dshokouhi** [comment]
Could you also attach a screenshot of the setting you created?  Just so we can see what it looks like.

**Comment by skynetua** [comment]
It is visible at light mode enabled
SmartSelect_20200811-234708_Home Assistant
SmartSelect_20200811-234747_Home Assistant

File: app/src/main/res/layout/fragment_mobile_app_integration.xml (hunk 1 above)
Comment (skynetua) [reply app/src/main/res/layout/fragment_mobile_app_integration.xml:152]: Oh, i've got what you mean after repository update)

**Pull request merged**